packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -1741,3 +1742,13 @@ class DevtoolsLauncher {
 
   static DevtoolsLauncher get instance => context.get<DevtoolsLauncher>() ?? DevtoolsLauncher();
 }
+
+class DevToolsServerAddress {
+  DevToolsServerAddress._(this.host, this.port, this.success);
+  DevToolsServerAddress.succeeded(String host, int port): this._(host, port, true);
+  DevToolsServerAddress.failed(): this._(null, null, false);
+
+  final String host;
+  final int port;
+  final bool success;
```

Comment:
Nit: if the class is called `DevToolsServerAddress` it seems weird that it contains `bool success`.

How about returning null on failure instead of an explicit field?

Reply:
That's true, I can make this a little simpler overall without the success/failure. I might add a failure message later on but I don't think it's needed now.